sources/bot/modules/spreadsheet/spreadsheet_handler.py

```diff
@@ -0,0 +1,165 @@
+"""
+"""
+import httplib2
+import apiclient
+from oauth2client.service_account import ServiceAccountCredentials
+from typing import List
+
+
+class SpreadsheetHandler:
+    def __init__(self, spreadsheet_id, file_name: str):
+        self._spreadsheet_id = spreadsheet_id
+        self._credentials_file = file_name
+
+        self._credentials = ServiceAccountCredentials.from_json_keyfile_name(
+            self._credentials_file,
+            ["https://www.googleapis.com/auth/spreadsheets", "https://www.googleapis.com/auth/drive"],
+        )
+        self._http_auth = self._credentials.authorize(httplib2.Http())
+        self._service = apiclient.discovery.build("sheets", "v4", http=self._http_auth)
+
+    def __create_student_sheet(self, row_count: int, column_count: int) -> None:
+        self._service.spreadsheets().batchUpdate(
+            spreadsheetId=self._spreadsheet_id,
+            body={
+                "requests": [
+                    {
+                        "addSheet": {
+                            "properties": {
+                                "title": "Студенты",
+                                "gridProperties": {"rowCount": row_count, "columnCount": column_count},
+                            }
+                        }
+                    }
+                ]
+            },
+        ).execute()
+
+        self._service.spreadsheets().values().batchUpdate(
+            spreadsheetId=self._spreadsheet_id,
+            body={
+                "valueInputOption": "USER_ENTERED",
+                "data": [
+                    {
+                        "range": "Студенты!A1:D1",
+                        "majorDimension": "ROWS",
+                        "values": [["username", "ФИО", "Группа", "Подруппа"]],
+                    }
```

Comment:
You need to implement spreadsheet handler without data binding (sheet names, coloumn names and other). After you need to create a class for entering student data, which will use previous class.

Reply:
Fixed